### PR TITLE
+ testkit Add java.time.Duration to testkit's javadsl

### PR DESCRIPTION
**Pull request by He-Pin**

### CODE REVIEW

File: akka-actor/src/main/scala/akka/util/Timeout.scala [hunk 1]
```diff
@@ -33,5 +33,13 @@ object Timeout {
    */
   def apply(length: Long, unit: TimeUnit): Timeout = new Timeout(length, unit)
 
+  /**
+   * Create an Timeout from java.time.Duration.
```
Comment (patriknw) [inline akka-actor/src/main/scala/akka/util/Timeout.scala]: a Timeout

File: akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala [hunk 2]
```diff
@@ -63,11 +64,26 @@ class TestKit(system: ActorSystem) {
     }
   }
 
+  /**
+   * Return an `Infinite` duration of Scala's `Duration`.
+   */
+  def infiniteDuration: Infinite = Duration.Inf
```
Comment (patriknw) [inline akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala]: doesn't `Duration.Inf()` work for Java?
strange to add more api for scala Duration when we are deprecating it in javadsl

File: akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala [hunk 3]
```diff
@@ -292,8 +568,20 @@ class TestKit(system: ActorSystem) {
...
+  @deprecated("Use the overloaded one which accepts java.time.Duration instead.", since = "2.5.12")
   def expectMsgAnyOf[T](max: FiniteDuration, objs: T*): T = tp.expectMsgAnyOf(max, objs: _*)
 
+  /**
+   * Receive one message from the test actor and assert that it equals one of
+   * the given objects. Wait time is bounded by the given duration, with an
+   * AssertionFailure being thrown in case of timeout.
+   */
+  def expectMsgAnyOf[T](max: java.time.Duration, objs: T*): T = {
+    import akka.util.JavaDurationConverters._
```
Comment (patriknw) [inline akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala]: would be ok to add this as an import at the top of the file to not have to do it at all places

**Review by patriknw — COMMENTED** [review]
looking good, some small nitpicks

File: akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala (hunk 2 above)
Comment (hepin1989) [reply akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala]: LoL, yes.

File: akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala (hunk 3 above)
Comment (hepin1989) [reply akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala]: OK:)

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Comment by patriknw** [comment]
Refs https://github.com/akka/akka/issues/24646

**Pull request merged**